src/System.Numerics.Experimental/Half.cs

```diff
@@ -353,5 +356,402 @@ public bool TryFormat(Span<char> destination, out int charsWritten, ReadOnlySpan
         {
             throw new NotImplementedException();
         }
+
+        // -----------------------Start of to-half conversions-------------------------
+
+        public static implicit operator Half(int i)
+        {
+            bool sign = i < 0;
+            Half h = (uint)(sign ? -i : i); // Math.Abs but doesn't throw exception, because we cast it to uint anyway
+            return sign ? new Half((ushort)(h.m_value | SignMask)) : h;
+        }
+
+        public static implicit operator Half(uint i)
+        {
+            if (i == 0)
+                return default;
+
+            int shiftDist = BitOperations.LeadingZeroCount(i) - 21;
+            if (shiftDist >= 0)
+                return new Half(false, (ushort)(0x18 - shiftDist), (ushort) (i << shiftDist));
+
+            shiftDist += 4;
+            uint sig = shiftDist < 0 ? ShiftRightJam(i, -shiftDist) : i << shiftDist;
+            return RoundPackToHalf(false, (short) (0x1C - shiftDist), (ushort) sig);
+        }
+
+        public static implicit operator Half(long l)
+        {
+            bool sign = l < 0;
+            Half h = (ulong)(sign ? -l : l); // Math.Abs but doesn't throw exception, because we cast it to ulong anyway
+            return sign ? new Half((ushort)(h.m_value | SignMask)) : h;
+        }
+
+        public static implicit operator Half(ulong l)
+        {
+            if (l == 0)
+                return default;
+
+            int shiftDist = BitOperations.LeadingZeroCount(l) - 53;
+
+            if (shiftDist >= 0)
+                return new Half(false, (ushort)(0x18 - shiftDist), (ushort)(l << shiftDist));
+
+            shiftDist += 4;
+            ushort sig = (ushort)(shiftDist < 0 ? ShiftRightJam(l, -shiftDist) : l << shiftDist);
+            return new Half(RoundPackToHalf(false, (short)(0x1C - shiftDist), sig));
+        }
+
+        public static implicit operator Half(short s)
+        {
+            return (int) s;
```

Comment:
note: These could probably be micro-optimized since we know the value will always fit (`byte`/`sbyte`) or that only a limited set of values will overflow (`short`/`ushort`)